src/lib/show-hide.spec.js

```diff
@@ -1,30 +1,227 @@
+/* eslint-disable noinspection, prefer-arrow-callback, func-names, no-underscore-dangle */
 import { show, hide, __RewireAPI__ } from './show-hide';
 
+const jsdom = require('jsdom-global');
+
 describe('show function', function () {
-  it('calls dispatchEvent on the wrapper with type `dialog:show-before` and dialog wrapper as data');
-  it('cancels show (does not call show-after or goToDialog ) if event was prevented', () => {
-    let myDispatch = sinon.stub().returns(false);
-    __RewireAPI__.__Rewire__('dispatchEvent', myDispatch);
-    let dialogDummy = { wrapper: 'test', goToDialog: sinon.spy() };
-    show.call(dialogDummy);
-    expect(myDispatch).to.have.been.calledOnce();
-    expect(myDispatch).to.have.been.calledWith('test', 'dialog:show-before', { dialog: 'test' });
-    expect(dialogDummy.goToDialog).to.have.been.not.called()
-  });
-  it('sets state to visible');
-  it('removes the aria-hidden attribute on the wrapper');
-  it('keeps track of focuses element on focusOnClose property on the dialogState ');
-  it('calls goToDialog(0)'); // either do this or continue the testing inwards - sometimes not practical...
-  it('sets aria-hidden on elemToConceal');
-  it('adds an eventListener on the document.body calling hideWhenFocusLost on focus ', () => {
-    let simulant = require('simulant');
-    let myDispatch = sinon.stub().returns(true);
+  let addEventListenerStub;
+  let myDispatch;
+  let dialogDummy;
+  let randomElementThatWasActive;
+  let cleanup;
+
+  beforeEach(function () {
+    cleanup = jsdom();
+  });
+
+  afterEach(function () {
+    cleanup();
+  });
+
+  it('should have access to document global', function () {
+    expect(document).to.not.be.undefined();
+  });
+  it('calls dispatchEvent on the wrapper with type `dialog:show-before` and dialog wrapper as data',
+    function () {
+      setup({ dispatchRetValue: false });
+      show.call(dialogDummy);
+      expect(myDispatch).to.have.been.calledOnce();
+      expect(myDispatch).to.have.been.calledWith(
+        dialogDummy.wrapper, 'dialog:show-before', { dialog: dialogDummy.wrapper });
+    });
+  it('cancels show (does not call show-after or goToDialog ) if event was prevented', function () {
+    setup({ dispatchRetValue: false });
+    expect(dialogDummy.goToDialog).to.have.been.not.called();
+    expect(myDispatch).to.not.have.been.calledWith(
+      dialogDummy.wrapper, 'dialog:show-before', { dialog: dialogDummy.wrapper });
+  });
+  it('sets state to visible', function () {
+    setup({ dispatchRetValue: true });
+    show.call(dialogDummy);
+    expect(dialogDummy.isVisible).to.be.true();
+  });
+  it('removes the aria-hidden attribute on the wrapper', function () {
+    setup({ dispatchRetValue: true });
+    show.call(dialogDummy);
+    expect(dialogDummy.wrapper.getAttribute('aria-hidden')).not.to.exist();
+  });
+  it('keeps track of focuses element on focusOnClose property on the dialogState ', function () {
```

Comment:
this test uses both show and hide, so it's place should be on neither describe blocks - it should be a third describe block for full flow tests.